### PR TITLE
UI/combobox

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Description 📣

## Type ✨

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [ ] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: frontend/src/components/v2/Modal/Modal.tsx
Comment (scott-ray-wilson) [inline frontend/src/components/v2/Modal/Modal.tsx:1]: The combobox can clip if the modal height is smaller than the list, preventing users from seeing selection
image

File: frontend/src/views/Project/MembersPage/components/IdentityTab/components/IdentityModal.tsx
Comment (scott-ray-wilson) [inline frontend/src/views/Project/MembersPage/components/IdentityTab/components/IdentityModal.tsx:1]: When I open the dialog there's no default selection for role. If I hit create without manually selecting one I get no error display.

image

Comment (scott-ray-wilson) [inline frontend/src/views/Project/MembersPage/components/IdentityTab/components/IdentityModal.tsx:1]: After adding an identity, if I reopen the dialog the combobox default value preserves the just added identity which will throw an error if submitted

Untitled

File: frontend/src/components/v2/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+import { faCaretDown, faCheck } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Combobox, Transition } from "@headlessui/react";
+import { ByComparator } from "@headlessui/react/dist/types";
+import { twMerge } from "tailwind-merge";
+
+type MyComboBoxProps<T extends object> = {
```
Comment (scott-ray-wilson) [inline frontend/src/components/v2/ComboBox/ComboBox.tsx:8]: maybe just `ComboBoxProps` not `MyComboBoxProps`?

File: frontend/src/components/v2/ComboBox/ComboBox.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  onFilter,
+  displayValue,
+  by,
+  items,
+  ...props
+}: MyComboBoxProps<T>) => {
+  const [query, setQuery] = useState("");
+
+  const filteredResult =
+    query === "" ? items.slice(0, 20) : items.filter((el) => onFilter(el, query));
```
Comment (scott-ray-wilson) [inline frontend/src/components/v2/ComboBox/ComboBox.tsx:36]: Should probably `slice` the filtered results as well? I have 10k identities and if I type it crashes the page

**Review by scott-ray-wilson — CHANGES_REQUESTED** [review]

File: frontend/src/components/v2/ComboBox/ComboBox.tsx (hunk 2 above)
Comment (maidul98) [reply frontend/src/components/v2/ComboBox/ComboBox.tsx:36]: Agree, we shouldn't show the whole list on the UI

File: frontend/src/components/v2/Modal/Modal.tsx
Comment (scott-ray-wilson) [reply frontend/src/components/v2/Modal/Modal.tsx:1]: addressed with `overflow-visible`

File: frontend/src/views/Project/MembersPage/components/IdentityTab/components/IdentityModal.tsx
Comment (scott-ray-wilson) [reply frontend/src/views/Project/MembersPage/components/IdentityTab/components/IdentityModal.tsx:1]: preventing submit until roles load and setting value once loaded

Comment (scott-ray-wilson) [reply frontend/src/views/Project/MembersPage/components/IdentityTab/components/IdentityModal.tsx:1]: setting reset value to next available identity after successful add

File: frontend/src/components/v2/ComboBox/ComboBox.tsx (hunk 2 above)
Comment (scott-ray-wilson) [reply frontend/src/components/v2/ComboBox/ComboBox.tsx:36]: adding `slice` drastically improved performance

**Review by scott-ray-wilson — APPROVED** [review]

**Pull request merged**